specification/sbi_atee.adoc

```diff
@@ -13,6 +44,953 @@ running a VMM).
+and the TEE workloads that run on it. This is accomplished by having a clear seperation between
+the non-confidential and confidential portions of the workloads running on the platform. The exact
+hardware and software mechanisms used to guarantee the isolation are platform dependent, but
+confidential workloads have absolute execution, data integrity, and confidentiality protections
+from amy component in the non-confidential world. The responsibility of the TSM is to enforce 
+the guarantees accorded to TEE workloads. Given its minimal footprint, most of the functionality
+for actually running the payloads, like memory allocation, scheduling, interrupt handling, 
+virtual devices is left to the OS/VMM (which are outside the TCB).
+
+In this scheme, most resources like memory, (V)CPUs start off as traditional untrusted resources
+in the non-confidential world, and are transtioned to their confidential analogue via the TSM.
+Once the conversion process is complete, a confidential-resource cannot be reverted to the 
+non-confidential world unless it's is no longer assigned to a TVM.
+
+The TEE address space can be comprised of confidential and non-confidential regions. The former includes
+both measured and code pages that are part of the TVM payload, and confidential zero-pages that can be
+mapped in by the VMM on demand. The non-confidential regions include non-confidential memory that can
+be used for virtual device emulation and so forth.
+
+=== Operational model
+
+Executing confidential workloads in a TEE requires a sequence of one or more of the steps detailed below.
+We'll assume that these steps are performed by an untrusted entity like the OS/VMM in conjunction with the
+TSM.
+
+. Platform TSM detection and capability enumeration
+. Conversion of non-confidential memory to confidential memory
+. Trusted VM (TVM) creation
+. Donating confidential memory to the TSM for TVM page management
+. Defining confidential memory regions
+. Defining MMIO regions
+. Defining non-confidential shared memory regions
+. Conversion of non-confidential (V)CPUs to confidential-VCPUs
+. Mapping code and data payloads for the TVM to confidential-memory regions
+. Mapping demand-zero confidential memory regions
+. Adding confidential VCPUs to the TVM
+. Finalizing TVM creation
+. Scheduling TVM execution
+. Management of secure interrupts
+. Handling and servicing TVM faults and exits
+. Mapping of confidential zero pages on demand
+. Mapping non-confidential shared pages on demand
+. Tearing down TVMs
+. Repurposing confidential memory for other TVMs
+. Reclaiming confidential memory for non-confidential VMs
+
+=== Platform TSM detection and capability enumeration
+
+Platform support for the TSM can be detected by probing for the EXT_TEE extension, and then
+calling sbi_tee_tsm_get_info() to get information about the current status of the TSM. The
+TSM must be be in TSM_READY in order to process further ECALLs.
+
```

Comment:
Describe the TSM states here in a table and point to that table in tsm_get_info ?